workbench/public/components/Main/main.tsx

```diff
@@ -260,13 +261,17 @@ export class Main extends React.Component<MainProps, MainState> {
       }
     }
     if (!response.data.ok) {
+      // Error message in JSON format from backend is wrapped in response data
+      // response.data: IHttpResponse<ResponseData> 
+      const error: IHttpResponse<ResponseData> = response.data;
       return {
         fulfilled: false,
-        errorMessage: response.data.resp,
-        data: response.data.body,
+        errorMessage: error.data.resp,
+        data: error.data.body,
       };
     }
 
+    // response: IHttpResponse<ResponseData> 
```

Comment:
Do we need to leave these comments in?

Reply:
Yes I added this comment here to remind us that the responses of success and error are of different types